packages/core/src/middleware/koa-consent-guard.ts

```diff
@@ -0,0 +1,71 @@
+import { experience } from '@logto/schemas';
+import { type MiddlewareType } from 'koa';
+import { type IRouterParamContext } from 'koa-router';
+import { errors, type Provider } from 'oidc-provider';
+
+import RequestError from '../errors/RequestError/index.js';
+import type Libraries from '../tenants/Libraries.js';
+import type Queries from '../tenants/Queries.js';
+import assertThat from '../utils/assert-that.js';
+
+/**
+ * Guard before allowing auto-consent
+ */
+export default function koaConsentGuard<
+  StateT,
+  ContextT extends IRouterParamContext,
+  ResponseBodyT,
+>(
+  provider: Provider,
+  libraries: Libraries,
+  query: Queries
+): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  return async (ctx, next) => {
+    const interactionDetails = await provider.interactionDetails(ctx.req, ctx.res);
+    const {
+      params: { token, login_hint: loginHint },
+      session,
+    } = interactionDetails;
+
+    assertThat(session, new errors.SessionNotFound('session not found'));
```

Comment:
use `RequestError('session.not_found')` ?

Reply:
This is consistent with the error handling in [existing code](https://github.com/logto-io/logto/blob/master/packages/core/src/routes/interaction/utils/interaction.ts#L156-L167).

Do we need to use `RequestError` instead now?

Reply:
For i18n support, I think RequestError is a better choice.

Reply:
OK, updated